lightning/src/ln/channelmanager.rs

```diff
@@ -1167,20 +1172,12 @@ impl<ChanSigner: ChannelKeys> ChannelManager<ChanSigner> {
 			return Ok(());
 		};
 
-		match handle_error!(self, err) {
-			Ok(_) => unreachable!(),
-			Err(e) => {
-				if let msgs::ErrorAction::IgnoreError = e.action {
```

Comment:
IIRC, that was quite intentional. Because you do want to log error even if it's a IgnoreError one but then was planning in further commit to match it out in the handle_error macro to avoid passing for nothing to peer_handler. I think that's good for now to pass it to peer_handler, it's just logged and ignored..

Reply:
No? The IgnoreErrors that are generated today are all things where you didn't already get back an APIError directly (as the user). There is no need to send duplicative errors (though logging them is, of course, fine IMO).

Reply:
I think there is already duplicative errors right now in send_payment/update_fee for the non-IgnoreError cases? Inside  handle_error, you don't know if an APIError is going to be returned, so for not generating a HandleError when  we're returning an APIError anyway I need to filter out all IgnoreError, and so modify test in consequence again ?